x-pack/plugins/security_solution/public/plugin_services.ts

```diff
@@ -112,6 +112,13 @@ export class PluginServices {
     });
     const customDataService = this.startCustomDataService(query, startPlugins.data);
 
+    const timelineDataService: DataPublicPluginStart = {
+      ...startPlugins.data,
```

Comment:
Do we need to include `...startPlugins.data` in `timelineDataService`? Seems that we are only using `timelineDataService.query.xxx`.

In line 124,  `...plugins` contains `DataPublicPluginStart` already

Reply:
Yep, we're passing the entire data service down to the `UnifiedFieldListSidebarContainer` as part of the `fieldListSidebarServices`.

Reply:
If we remove line 116 in the file, and keep the query; In `x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/index.tsx`  we take `data` from useKibana and `query` as an object, pass down to `fieldListSidebarServices` we could avoid duplicated return in here.

Reply:
I would prefer creating the service at this level though, so that none of that logic ends up in components. And I just moved it into its own setup function: https://github.com/elastic/kibana/pull/180850/commits/a01bb4a6d9053e918a8cf48623d3b23b97d7e641

Reply:
Interesting point @angorayc, However, we would have to keep an incomplete `timelineDataService` in the services context, with `query` alone, and inject the `...startPlugins.data` in the component logic before using it in the _fieldListSidebarServices_, which could be confusing.

To keep all the logic in the same place, we could also consider setting `query` as a service in the context, and build the entire `timelineDataService` in the components, something like:
```
const useTimelineDataService = () => {
    const { data, query, timelineFilterManager } = useKibana().services;
    return { ...data, query: { ...query, filterManager: timelineFilterManager }}; // useMemo
};
```

Nevertheless, I think the current approach is okay, we keep a duplicated copy of the top-level object from the `data` service, the rest are references. And the way to retrieve the global `filterManager` from the `data` service, and the timeline `filterManager` from the `timelineDataService` is the same, which makes a lot of sense. It looks clean to me.